File: 2020/01/main_test.go
package main

import "testing"

func TestTask1ShouldFindResult(t *testing.T) {
	input := []int{
		1721,
		979,
		366,
		299,
		675,
		1456,
	}
	expect := 514579

	result := Task1(input)
	if result != expect {
		t.Errorf("got %d, expected %d", result, expect)
	}
}

func TestTask2ShouldFindResult(t *testing.T) {
	input := []int{
		1721,
		979,
		366,
		299,
		675,
		1456,
	}
	expect := 241861950

	result := Task2(input)
	if result != expect {
		t.Errorf("got %d, expected %d", result, expect)
	}
}

File: 2021/02/src/main_test.zig
const std = @import("std");

const app = @import("./main.zig");

test "expect task 1 to yield correct result" {
    const input = [_]app.Command{
        .{ .cmd = .forward, .value = 5 },
        .{ .cmd = .down, .value = 5 },
        .{ .cmd = .forward, .value = 8 },
        .{ .cmd = .up, .value = 3 },
        .{ .cmd = .down, .value = 8 },
        .{ .cmd = .forward, .value = 2 },
    };
    const expected: i32 = 150;

    try std.testing.expect(app.task1(input[0..]) == expected);
}

test "expect task 2 to yield correct result" {
    const input = [_]app.Command{
        .{ .cmd = .forward, .value = 5 },
        .{ .cmd = .down, .value = 5 },
        .{ .cmd = .forward, .value = 8 },
        .{ .cmd = .up, .value = 3 },
        .{ .cmd = .down, .value = 8 },
        .{ .cmd = .forward, .value = 2 },
    };
    const expected: i32 = 900;

    try std.testing.expect(app.task2(input[0..]) == expected);
}
